grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -386,35 +386,25 @@ class OrderController {
     }
 
     def deleteAdjustment = {
-        def orderInstance = Order.get(params.order.id)
-        if (!orderInstance) {
+        User user = User.get(session?.user?.id)
+
+        Order order = Order.get(params.order.id)
+        if (!order) {
             flash.message = "${warehouse.message(code: 'default.not.found.message', args: [warehouse.message(code: 'order.label', default: 'Order'), params.order.id])}"
             redirect(action: "list")
-        } else {
-            User user = User.get(session?.user?.id)
-            def canEdit = orderService.canManageAdjustments(orderInstance, user)
-            if(canEdit) {
-                def orderAdjustment = OrderAdjustment.get(params?.id)
-                if (!orderAdjustment) {
-                    flash.message = "${warehouse.message(code: 'default.not.found.message', args: [warehouse.message(code: 'orderAdjustment.label', default: 'Order Adjustment'), params.id])}"
-                    redirect(action: "show", id: orderInstance?.id)
-                } else {
-                    if (orderAdjustment.hasRegularInvoice) {
-                        throw new UnsupportedOperationException("${warehouse.message(code: 'errors.noPermissions.label')}")
-                    }
-                    orderInstance.removeFromOrderAdjustments(orderAdjustment)
-                    orderAdjustment.delete()
-                    if (!orderInstance.hasErrors() && orderInstance.save(flush: true)) {
-                        flash.message = "${warehouse.message(code: 'default.updated.message', args: [warehouse.message(code: 'order.label', default: 'Order'), orderInstance.id])}"
-                        redirect(controller:"purchaseOrder", action: "addItems", id: orderInstance.id, params:['skipTo': 'adjustments'])
-                    } else {
-                        render(view: "show", model: [orderInstance: orderInstance])
-                    }
-                }
-            } else {
-                throw new UnsupportedOperationException("${warehouse.message(code: 'errors.noPermissions.label')}")
-            }
         }
+
+        OrderAdjustment orderAdjustment = OrderAdjustment.get(params?.id)
+        if (!orderAdjustment) {
+            flash.message = "${warehouse.message(code: 'default.not.found.message', args: [warehouse.message(code: 'orderAdjustment.label', default: 'Order Adjustment'), params.id])}"
+            redirect(action: "show", id: params.order.id)
+        }
+
+        orderService.deleteAdjustment(order, orderAdjustment, user)
```

Comment:
I believe you don't have to pass order here, because you can make `orderAdjustment.order`